### DISCUSSION TITLE
`--password-same` not working as expected when using `--user pi`

### DISCUSSION THREAD

**Discussion by tiefenauer** [gitbls/sdm#55] [Q&A]
TLDR: I have the following problems or things I don't understand:
1. correct usage of `--password-same`
1. redefining user `pi` with `--user`
1. `--autologin` does not seem tow work

I'm trying to set up a RaspberryPi while keeping the default user `pi` and using `pi` as the password for `pi` as well as for root. At the moment I set the following switches
```
--user pi --password-user pi --password-pi pi --rootpwd --autologin 
```
This works. However, there seems to be a little shorter variant with `--password-same`. I tried the following:

```
--user pi --password-pi pi --rootpwd --password-same y --autologin 
```

This does not work, as I'm prompted to enter a password manually (although I already set it with `--password-pi`):
```
** Enter password to use for users 'pi' and 'root': 
```

Also, the switches for the users/passwords are a bit confusing for me. `--user` is mandatory, even if I don't want to create a new user (only re-using the existing user `pi`). It should be made optional, IMO, only `--password-pi` should be mandatory.

Furthermore, `--autologin` does not seem to work (or I'm doing something wrong), as I'm still prompted for the password 

My goal is a pre-configured RaspbianOS with only two users: `pi` and `root`. Both users should have the same password and `pi` should be logged in automatically after startup.

**Comment by gitbls** [comment]
I assume this is on Bullseye 09/22? I was aware that there was an issue with passwords and the 'pi' user, but my assumption (hope?) was that people wouldn't use the user 'pi' any more, since it's slightly stuffed into the background by the RasPiOS team.

What IMG are you using? Desktop or Lite?

`--autologin` should work. I'll have a look into that.  Please post /etc/sdm/history from a run with `--autologin` requested that doesn't work correctly when booted. 

Will look at 'pi' user if you're locked into using it in your project. LMK.

**Comment by gitbls** [comment]
WRT `--autologin` not working: Are you using `--reboot` or `--restart`? If you are not, I can see that there could be an issue with autologin not getting set correctly, but won't investigate further until I hear from you on this.

**Reply by tiefenauer** [reply]
I'm not using neither the reboot nor the restart switch. I will post my /etc/sdm/history end of week when I'm home

**Comment by gitbls** [comment]
OK, thanks. Don't need to see your history file now, I understand what's going on.

**Comment by gitbls** [comment]
JFYI the next update will address:
- Autologin works correctly in all cases
- If no `--user` specified, defaults to **pi** 
- As a result, can now use `--password-pi` without specifying `--user pi`

I think this addresses the points you've raised here.

Coming later this week.
